### PR TITLE
mon: post-luminous cleanup (part 3 of ?)

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-13_13:31:57-rados-wip-sage-testing-2017-09-12-1750-distro-basic-smithi/

**Comment by liewegas** [comment]
will rebase

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -1776,7 +1776,8 @@ void Monitor::handle_probe_reply(MonOpRequestRef op)
 	   << " upgrading" << dendl;
       exit(0);
     }
-    if (!osdmon()->osdmap.test_flag(CEPH_OSDMAP_PURGED_SNAPDIRS)) {
+    if (!osdmon()->osdmap.test_flag(CEPH_OSDMAP_PURGED_SNAPDIRS) ||
+	!osdmon()->osdmap.test_flag(CEPH_OSDMAP_RECOVERY_DELETES)) {
```
Comment (gregsfortytwo) [inline src/mon/Monitor.cc]: Is RECOVERY_DELETES actually set as part of a scrub? I don't know how that works.

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -7481,16 +7474,6 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       goto reply;
     }
     if (prefix == "osd crush weight-set create") {
-      if (osdmap.require_min_compat_client > 0 &&
-	  osdmap.require_min_compat_client < CEPH_RELEASE_LUMINOUS) {
```
Comment (gregsfortytwo) [inline src/mon/OSDMonitor.cc]: Surely we don't mean to require Luminous clients to talk to Mimic clusters?

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -9261,21 +9223,6 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
              prefix == "osd rm-pg-upmap" ||
              prefix == "osd pg-upmap-items" ||
              prefix == "osd rm-pg-upmap-items") {
-    if (osdmap.require_osd_release < CEPH_RELEASE_LUMINOUS) {
-      ss << "you must complete the upgrade and 'ceph osd require-osd-release "
-	 << "luminous' before using the new interface";
-      err = -EPERM;
-      goto reply;
-    }
-    if (osdmap.require_min_compat_client < CEPH_RELEASE_LUMINOUS) {
```
Comment (gregsfortytwo) [inline src/mon/OSDMonitor.cc]: Again, client.

File: src/mon/Monitor.cc (hunk 1 above)
Comment (liewegas) [reply src/mon/Monitor.cc]: 807c4c834136e5a3d20b31d50e770430695473ff

File: src/mon/OSDMonitor.cc (hunk 3 above)
Comment (liewegas) [reply src/mon/OSDMonitor.cc]: good catch, fixed!

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-18_23:57:25-rados-wip-sage-testing-2017-09-18-1640-distro-basic-smithi/
http://pulpito.ceph.com/sage-2017-09-19_12:22:41-rados-wip-sage-testing-2017-09-18-1640-distro-basic-smithi/

**Comment by liewegas** [comment]
@gregsfortytwo ok to merge?

**Comment by gregsfortytwo** [comment]
jenkins test docs

**Comment by gregsfortytwo** [comment]
Yep, looks good @liewegas!

**Comment by liewegas** [comment]
thanks!

**Pull request merged**